docs/developer.qmd

```diff
@@ -1,4 +1,270 @@
+First we will add the new name:
+
+```py
+ui_tabs_lookup = {
+	'tabBrowser': self.setupTabBrowser,
+   	...
+   	'tabInfo': self.setupTabInfo,
+	'tabExample': self.setupTabExample
+}
+```
+
+Now we need to write the function `setupTabExample`
+Add the new function below `setupTabBrowser`:
+
+```py
+    def setupTabExample(self):
+        self.exampleTab = gui.ExampleTab.ExampleTab()
+        self.tabWidget.addTab(self.exampleTab, 'Example')
+```
+Don't forget to add the new keyword to the tabs in your *ibridges_config.json*
+
+Now the new tab should appear as last tab in the application and we can continue definning its behaviour.
+:::
+
+::: {.callout-note collapse="true"}
+### 5. Defining the functionality of the new tab 
+
+You migt have observed that we already defined a variable `context` as a global variable. The class `utils.context.Context` contains all parameters from the *irods_environment.json*, *ibridges_config.json* and it contains an object `context.irods_connector`.
+
+```py
+import utils
+context = utils.context.Context()
+dir(context)
+
+[
+ 'application_name',
+ 'ibridges_conf_file',
+ 'ibridges_configuration',
+ 'ienv_is_complete',
+ 'irods_connector',
+ 'irods_env_file',
+ 'irods_environment',
+ 'reset',
+ 'save_ibridges_configuration',
+ 'save_irods_environment']
+```
+The class `Context` is a singleton, i.e. no matter how often you instantiate a `Context` object
+ there will just be one object in memory.
+
+```py
+context = utils.context.Context()
+context1 = utils.context.Context()
+
+context == context1
+True
+``` 
+
+That also means that if you change the values in the object, all other classes using `Context` will have the same information.
+
+`irods_connector` contains all functionality to interact with the iRODS server.
+
+With all that you can define the functionality of the tab, e.g.:
+
+```py
+import PyQt6.QtCore
+import PyQt6.QtGui
+import PyQt6.QtWidgets
+import PyQt6.uic
+
+import gui
+import meta
+import utils
+from gui.irodsTreeView import IrodsModel
+
+class IrodsExampleTab(PyQt6.QtWidgets.QWidget,
+                   gui.ui_files.ExampleTab.Ui_ExampleTab):
+    context = utils.context.Context()
+    def __init__(self):
+        super().__init__()
+        if getattr(sys, 'frozen', False):
+            super().setupUi(self)
+        else:
+            PyQt6.uic.loadUi("gui/ui_files/ExampleTab.ui", self)
+        self.error_label.setText("Whooohoo")
+
+        self._initialize_irods_model(self.irodsTreeView)
+        self.irodsTreeView.clicked.connect(self.treeFunction)
+
+    def _initialize_irods_model(self, treeView):
+        self.irodsmodel = IrodsModel(treeView)
+        treeView.setModel(self.irodsmodel)
+        irodsRootColl = self.context.irods_environment.config.get(
+                'irods_home', '/'+self.context.irods_connector.zone)
```

Comment:
Perhaps some explanation of the `path` module and how it is used would be ideal.  For instance, these two lines could be rewritten as an example:

```python
        ienv_dict = self.context.irods_environment.config
        home_coll_str = utils.path.IrodsPath(
            '/', self.context.irods_connector.zone, 'home')
        irods_root_coll = ienv_dict.get('irods_home', home_coll_str)
```

Reply:
Done! Thanks. I also added a small section to illustrate the use of IrodsPath.